src/main/java/com/bootcamp/demo/business/AccountService.java

```diff
@@ -154,14 +152,45 @@ public void update(AccountDetails account) throws ServiceException {
             try {
                 repository.update(transformedAccount);
             } catch (InterruptedException | ExecutionException e) {
-                throw new ServiceException(ErrorCode.INTERNAL, e.getMessage());
+                throw new ServiceException(ServiceException.ErrorCode.INTERNAL, e.getMessage());
             }
             catch(Exception e){
-                throw new ServiceException(ErrorCode.VALIDATION, e.getMessage());
+                throw new ServiceException(ServiceException.ErrorCode.VALIDATION, e.getMessage());
             }
         } else {
-            throw new ServiceException(ErrorCode.VALIDATION, validationResponse.getMessages().toString());
+            throw new ServiceException(ServiceException.ErrorCode.VALIDATION, validationResponse.getMessages().toString());
+        }
+    }
+
+    public void addPromotionToAccount(String accountId, String code) throws ServiceException {
+        try{
+            Optional<Promotion> promotionOptional = this.promotionRepository.popPromotion(code);
+            if(promotionOptional.isEmpty()){
+                throw new ServiceException(ServiceException.ErrorCode.VALIDATION, "Invalid code");
+            }
+            this.repository.addPromotionToAccount(accountId, promotionOptional.get());
+        } catch (ExecutionException | InterruptedException e) {
+            throw new ServiceException(ServiceException.ErrorCode.INTERNAL, e.getMessage());
+        }
+        catch(Exception exception)
+        {
+            throw new ServiceException(ServiceException.ErrorCode.VALIDATION, exception.getMessage());
+        }
+    }
+
+    public List<Promotion> getAccountPromotions(String accountId) throws ServiceException{
```

Comment:
minor formatting issue, missing space. U could use Intellij shortcut for formatting